src/ast/passes/codegen_llvm.cpp

```diff
@@ -3525,6 +3525,12 @@ libbpf::bpf_map_type CodegenLLVM::get_map_type(const SizedType &val_type,
               val_type.IsMinTy() || val_type.IsMaxTy() || val_type.IsAvgTy() ||
               val_type.IsStatsTy())) {
     return libbpf::BPF_MAP_TYPE_PERCPU_HASH;
+  } else if (!(val_type.IsHistTy() || val_type.IsLhistTy() ||
+               val_type.IsAvgTy() || val_type.IsStatsTy() ||
+               val_type.IsCountTy() || val_type.IsSumTy() ||
+               val_type.IsMaxTy() || val_type.IsMinTy()) &&
```

Comment:
This list is the same as above. Could we enclose the check into a function and use it for both?

For example a function called `needs_percpu_map` (or similar) sounds reasonable.